src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/AutoConfirmUser/AutomaticallyConfirmOrganizationUsersValidator.cs

```diff
@@ -0,0 +1,120 @@
+    ) : IAutomaticallyConfirmOrganizationUsersValidator
+{
+    public async Task<ValidationResult<AutomaticallyConfirmOrganizationUserRequestData>> ValidateAsync(
+        AutomaticallyConfirmOrganizationUserRequestData request)
+    {
+        var orgUser = request.OrganizationUser;
+
+        if (orgUser.Status != OrganizationUserStatusType.Accepted)
+        {
+            return Invalid(request, new UserIsNotAccepted());
+        }
+
+        if (orgUser.OrganizationId != request.Organization.Id)
+        {
+            return Invalid(request, new OrganizationUserIdIsInvalid());
+        }
+
+        return await OrganizationUserOwnershipValidationAsync(request)
+            .ThenAsync(OrganizationUserConformsToOrganizationPoliciesAsync);
+    }
+
+    private async Task<ValidationResult<AutomaticallyConfirmOrganizationUserRequestData>> OrganizationUserOwnershipValidationAsync(
+        AutomaticallyConfirmOrganizationUserRequestData request) =>
+        request.Organization.PlanType is not PlanType.Free
+        || await organizationUserRepository.GetCountByFreeOrganizationAdminUserAsync(request.UserId) == 0
+            ? Valid(request)
+            : Invalid(request, new UserToConfirmIsAnAdminOrOwnerOfAnotherFreeOrganization());
+
+    private async Task<ValidationResult<AutomaticallyConfirmOrganizationUserRequestData>>
+        OrganizationUserConformsToOrganizationPoliciesAsync(AutomaticallyConfirmOrganizationUserRequestData request)
+    {
+        return await OrganizationUserConformsToTwoFactorRequiredPolicyAsync(request)
+            .ThenAsync(OrganizationUserConformsToSingleOrgPolicyAsync);
+    }
+
+    private async Task<ValidationResult<AutomaticallyConfirmOrganizationUserRequestData>> OrganizationUserConformsToTwoFactorRequiredPolicyAsync(
+            AutomaticallyConfirmOrganizationUserRequestData request)
+    {
+        if ((await twoFactorIsEnabledQuery.TwoFactorIsEnabledAsync([request.UserId]))
+            .Any(x => x.userId == request.UserId && x.twoFactorIsEnabled))
+        {
+            return Valid(request);
+        }
+
+        if (!featureService.IsEnabled(FeatureFlagKeys.PolicyRequirements))
+        {
+            return (await policyService.GetPoliciesApplicableToUserAsync(request.UserId,
+                    PolicyType.TwoFactorAuthentication))
+                .Any(p => p.OrganizationId == request.Organization.Id)
+                    ? Invalid(request, new UserDoesNotHaveTwoFactorEnabled())
+                    : Valid(request);
+        }
+
+        return (await policyRequirementQuery.GetAsync<RequireTwoFactorPolicyRequirement>(request.UserId))
+            .IsTwoFactorRequiredForOrganization(request.Organization.Id)
+                ? Invalid(request, new UserDoesNotHaveTwoFactorEnabled())
+                : Valid(request);
+    }
+
+    private async Task<ValidationResult<AutomaticallyConfirmOrganizationUserRequestData>> OrganizationUserConformsToSingleOrgPolicyAsync(
```

Comment:
Does this remove the single org exemption for owners and admins? This probably needs a bit of thought, depending on whether you use `policyService` or `policyRequirement`s. At the moment I think both code paths maintain the exemption. I think `PolicyRequirement` gives you slightly finer grained control over this without affecting other policies.